gobblin-modules/gobblin-orc/src/main/java/org/apache/gobblin/writer/GobblinOrcWriterBuilder.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.gobblin.writer;
+
+import java.io.IOException;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+
+
+public class GobblinOrcWriterBuilder extends FsDataWriterBuilder<Schema, GenericRecord> {
+  public GobblinOrcWriterBuilder() {
+  }
+
+  @Override
+  public DataWriter<GenericRecord> build()
+      throws IOException {
+    Preconditions.checkNotNull(this.destination);
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(this.writerId));
+    Preconditions.checkNotNull(this.schema);
+
+    switch (this.destination.getType()) {
+      case HDFS:
```

Comment:
Just an FYI - there have been questions on Gitter whether Gobblin supports writing in ORC format to S3. What aspects of GobblinOrcWriter can be reused for future writes to S3?

Reply:
Mostly it should be compatible. users will need to be cautious on the staging directory, where the rename operation on `s3a` is not a metadata operation but triggers a data copy. So it will be more efficient to set the staging directory of writer as a local destination and publish to s3a in publisher.